java-checks/src/main/java/org/sonar/java/checks/serialization/SerializableSuperConstructorCheck.java

```diff
@@ -60,7 +60,10 @@ public void visitNode(Tree tree) {
   }
 
   private static boolean isNotSerializableMissingNoArgConstructor(@Nullable Type superclass) {
-    return superclass != null && !isSerializable(superclass) && !hasNonPrivateNoArgConstructor(superclass);
+    return superclass != null
```

Comment:
this is the uncovered condition that was already uncovered, I believe it is there for defensive programming because it is always going to be not null at this point.

Reply:
Indeed, it looks like we always get a `Type` object. Let's keep it for safety then.